.github/workflows/release.yml

```diff
@@ -1,63 +1,50 @@
-name: Deploy release candidate
+name: Release
 on:
     push:
         branches:
             - main
     pull_request: # Runs on pull requests to any branch
 env:
-    IS_RELEASE_CANDIDATE: ${{ contains(github.head_ref, 'release--') }}
-    IS_FEATURE: ${{ !contains(github.head_ref, 'release--') }}
     DFX_VERSION: 0.21.0
     NODE_VERSION: 20
 jobs:
-    get-tests:
+    determine-should-release:
         if: ${{ contains(github.head_ref, 'release--') }}
+        name: Determine if this branch should release
+        runs-on: ubuntu-latest
+        outputs:
+            should_release: ${{ steps.determine_should_release.outputs.should_release }}
+        steps:
+            - uses: actions/checkout@v4
+
+            - id: determine_should_release
+              uses: ./.github/actions/should_release
+
+    get-tests:
+        needs: determine-should-release
+        if: ${{ contains(github.head_ref, 'release--') && needs.determine-should-release.outputs.should_release }}
```

Comment:
Can you give this a name? The release-- thing? Can you reuse the same variable location as in the determine_should-release action?

Reply:
I don't think it's worth the overhead. When we finalize the review I can show you what it would take

But we would either have to 1. dilute an already pretty concise job with an output that doesn't really fit (and is actually required by the job, or 2. make a brand new job just for this one variable. That job will have to run to completion before any of the other jobs can start. 

I understand that the desire to do this, and I tried many times to make this happen but ultimately decided that it wasn't worth the trouble. It is used very infrequently, and is reasonably straight forward as is